lib/internal/streams/readable.js

```diff
@@ -1062,8 +1062,17 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
-  let stream = this;
+  return streamToAsyncIterator(this);
+};
+
+Readable.prototype.iterator = function(options = {
+  destroyOnReturn: true,
+  destroyOnError: true,
+}) {
```

Comment:
Setting the default in this way does not work as you you'd expect. Passing passing an empty object does not set them to true.

Reply:
That's why I destructured  them as I did in `streamToAsyncIterator`

Reply:
Let's handle them once inside `streamToAsyncIterator`.